dashboard/src/app/get-started/template-list/template-list.controller.ts

```diff
@@ -118,31 +118,43 @@ export class TemplateListController {
       this.$log.error(message);
       return;
     }
+
+    const urls = this.devfileRegistryUrl.split(" ");
+    let promises = [];
+   
     this.isLoading = true;
-    this.devfileRegistry.fetchDevfiles(this.devfileRegistryUrl).then((devfiles: Array<IDevfileMetaData>) => {
-      this.devfiles = devfiles.map(devfile => {
-        if (!devfile.icon.startsWith('http')) {
-          devfile.icon = this.devfileRegistryUrl + devfile.icon;
+      
+    for (const url of urls) {
+      promises.push(this.devfileRegistry.fetchDevfiles(url).then((devfiles: Array<IDevfileMetaData>) => {
```

Comment:
This logic with separating registry URLs and changing devfile icons URLs should be in one place.

I think the better solution to move it into  DevfileRegistry class. "fetchDevfiles" should return all needed objects. 
https://github.com/eclipse/che/blob/master/dashboard/src/components/api/devfile-registry.factory.ts#L65